test/functionality/abstract-interpretation/data-frame/data-frame.ts

```diff
@@ -32,56 +32,74 @@ export const DataFrameTestOverapproximation = {
 	rows:     DomainMatchingType.Overapproximation
 };
 
+interface CriterionTestEntry {
+	criterion:  SingleSlicingCriterion,
+	value:      DataFrameDomain,
+	node:       RSymbol<object>,
+	lineNumber: number
+}
+
 export function assertDataFrameDomain(
 	shell: RShell,
 	code: string,
-	criterion: SingleSlicingCriterion,
-	expected: DataFrameDomain,
+	expected: [SingleSlicingCriterion, DataFrameDomain][],
 	name: string = code
 ) {
-	test(name, async()=> {
+	test.each(expected)(name, async(criterion, expect) => {
 		const [value] = await getInferredDomainForCriterion(shell, code, criterion);
```

Comment:
This can be probably outsourced into another issue, but given that everything but requesting the result for the criterion is independent, we could use a `beforeAll` to parse & analyze the code once and then only check each of the criteria.